mautic/mautic
# Fix Invalid clickthrough value exception

| Q                                      | A
| -------------------------------------- | ---
| Branch?                                | "features" for all features, enhancements and bug fixes (until 3.3.0 is released) 
| Bug fix?                               | yes/no
| New feature?                           | yes/no
| Deprecations?                          | yes/no
| BC breaks?                             | yes/no
| Automated tests included?              | yes/no
| Related user documentation PR URL      | mautic/mautic-documentation#... 
| Related developer documentation PR URL | mautic/developer-documentation#... 
| Issue(s) addressed                     | Fixes #9400




#### Description:
This PR fix issue millions logs like:

`[2021-09-02 10:28:41] mautic.ERROR: Invalid clickthrough value: Yjo4OntmOjk6ImZiaGVkZiI7Yjo1Ont2OjM7Zjo4OiJmemJ2eSI7djo0O3Y6NDAyO31mOjg6ImZ6YnZ5Ijt2OjQwMjtmOjc6ImZnYmciO2Y6NTU6Ijk0NjNiNWNjMWRmNzk5MjcwNjI0NDUiO2Y6NzoieWZiZSI7ZjowOiI0OTY1Nzg1IjtmOjA6ImR1YmFhZnkiO2I6NDp7Zjo4OiJmemJ2eSI7djo0MDI7fX0= {"exception":"[object] (Mautic\\CoreBundle\\Exception\\InvalidDecodedStringException(code: 0): The string b:8:{f:9:\"fbhedf\";b:5:{v:3;f:8:\"fzbvy\";v:4;v:402;}f:8:\"fzbvy\";v:402;f:7:\"fgbg\";f:55:\"9463b5cc1df79927062445\";f:7:\"yfbe\";f:0:\"4965785\";f:0:\"dubaafy\";b:4:{f:8:\"fzbvy\";v:402;}} is not a serialized array at..`

I collect the requests, and it looks like this kind of requests are from bots (Microsoft....)
Then we should skip it.



#### Steps to test this PR:
1. Load up [this PR](https://mautibox.com)
2. Go to your page with tracking code
3. Add to url `?ct=Yjo4OntmOjk6ImZiaGVkZiI7Yjo1Ont2OjM7Zjo4OiJmemJ2eSI7djo0O3Y6NTM4O31mOjg6ImZ6YnZ5Ijt2OjUzODtmOjc6ImZnYmciO2Y6NTU6Ijk0NzY4Y2I4YmZhMGY4Nzc2NTMzODEiO2Y6NzoieWZiZSI7Zjo5OiI4MjM5OTgiO2Y6MDoiZHViYWFmeSI7Yjo0OntmOjg6ImZ6YnZ5Ijt2OjUzODt9fQ`
4. Check your logs and you should see error :  `mautic.ERROR: Invalid clickthrough value: `
5. After PR page you should see not tracked page in the browser console - we asume it's bot

![image](https://user-images.githubusercontent.com/462477/133642035-0df7fb45-59d5-443e-9b21-abb2eb29e04c.png)
